opts/ip.go

```diff
@@ -5,20 +5,22 @@ import (
 	"net"
 )
 
-// IpOpt type that hold an IP
-type IpOpt struct {
+// IPOpt type that hold an IP
+type IPOpt struct {
 	*net.IP
 }
 
-func NewIpOpt(ref *net.IP, defaultVal string) *IpOpt {
-	o := &IpOpt{
+// NewIPOpt returns a new IPOpt from a string of a IP
+func NewIPOpt(ref *net.IP, defaultVal string) *IPOpt {
+	o := &IPOpt{
 		IP: ref,
 	}
 	o.Set(defaultVal)
 	return o
 }
 
-func (o *IpOpt) Set(val string) error {
+// Set validates a given string of a IPv4 or IPv6
```

Comment:
"Set sets an IPv4 or IPv6 address from a given string. If the given string is not parsable as an IP address it will return an error"